zookeeper-server/src/test/java/org/apache/zookeeper/test/AuthTest.java

```diff
@@ -20,22 +20,38 @@
 
 import static org.junit.Assert.fail;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.TestableZooKeeper;
 import org.apache.zookeeper.WatchedEvent;
 import org.apache.zookeeper.Watcher.Event.KeeperState;
+import org.apache.zookeeper.ZooDefs;
 import org.apache.zookeeper.ZooDefs.Ids;
 import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Id;
+import org.apache.zookeeper.server.auth.DigestStrategyFactory;
 import org.junit.Test;
 
 public class AuthTest extends ClientBase {
 
     static {
         // password is test
-        System.setProperty("zookeeper.DigestAuthenticationProvider.superDigest", "super:D/InIHSb7yEEbrWz8b9l71RjZJU=");
+        if (System.currentTimeMillis() % 3 == 0) {
```

Comment:
I don't understand this part.
You should run the tests against all digest types

Reply:
I redo the unit test